data/org.sugarlabs.gschema.xml

```diff
@@ -333,4 +333,177 @@
+    <enum id='org.sugarlabs.desktop.GDesktopProxyMode'>
+        <value nick='none' value='0'/>
+        <value nick='manual' value='1'/>
+        <value nick='auto' value='2'/>
+        <value nick='system' value='3'/>
+    </enum>
+    <schema id="org.sugarlabs.system.proxy" path="/org/sugarlabs/system/proxy/">
+        <child name="http" schema="org.sugarlabs.system.proxy.http"/>
+        <child name="https" schema="org.sugarlabs.system.proxy.https"/>
+        <child name="ftp" schema="org.sugarlabs.system.proxy.ftp"/>
+        <child name="socks" schema="org.sugarlabs.system.proxy.socks"/>
+        <key name="mode" enum="org.sugarlabs.desktop.GDesktopProxyMode">
+            <default>'none'</default>
+            <summary>Proxy configuration mode</summary>
+            <description>
+            Select the proxy configuration mode. Supported values are "none", 
+            "manual", "auto".
+
+            If this is "none", then proxies are not used.
+
+            If it is "auto", the autoconfiguration URL described by the
+            "autoconfig-url" key is used.
+
+            If it is "manual", then the proxies described by
+            "/system/proxy/http", "/system/proxy/https",
+            "/system/proxy/ftp" and "/system/proxy/socks" will be used.
+            Each of the 4 proxy types is enabled if its "host" key is
+            non-empty and its "port" key is non-0.
+
+            If an http proxy is configured, but an https proxy is not,
+            then the http proxy is also used for https.
+
+            If a SOCKS proxy is configured, it is used for all protocols,
+            except that the http, https, and ftp proxy settings override
+            it for those protocols only.
+            </description>
+        </key>
+        <key name="autoconfig-url" type="s">
+            <default>''</default>
+            <summary>Automatic proxy configuration URL</summary>
+            <description>
+            URL that provides proxy configuration values. When mode is
+            "auto", this URL is used to look up proxy information for all
+            protocols.
+            </description>
+        </key>
+        <key name="ignore-hosts" type="as">
+            <default>[ 'localhost', '127.0.0.0/8', '::1' ]</default>
+            <summary>Non-proxy hosts</summary>
+            <description>
+            This key contains a list of hosts which are connected to directly, 
+            rather than via the proxy (if it is active). The values can be 
+            hostnames, domains (using an initial wildcard like *.foo.com), IP host 
+            addresses (both IPv4 and IPv6) and network addresses with a netmask 
+            (something like 192.168.0.0/24).
+            </description>
+        </key>
+        <key name="use-same-proxy" type="b">
+            <default>true</default>
+            <summary>Unused; ignore</summary>
+            <description>
+            This key is not used, and should not be read or modified.
```

Comment:
use-same-proxy is a short way of saying use the same proxy for all other
protocals like https ftp and should auto populate the remaining proxy boxes
for those services.

On Sun, May 8, 2016 at 5:37 AM, Sam <notifications@github.com> wrote:

> In data/org.sugarlabs.gschema.xml
> <https://github.com/sugarlabs/sugar/pull/665#discussion_r62429966>:
>
> > +        <key name="ignore-hosts" type="as">
> > +            <default>[ 'localhost', '127.0.0.0/8', '::1' ]</default>
> > +            <summary>Non-proxy hosts</summary>
> > +            <description>
> > +            This key contains a list of hosts which are connected to directly,
> > +            rather than via the proxy (if it is active). The values can be
> > +            hostnames, domains (using an initial wildcard like *.foo.com), IP host
> > +            addresses (both IPv4 and IPv6) and network addresses with a netmask
> > +            (something like 192.168.0.0/24).
> > +            </description>
> > +        </key>
> > +        <key name="use-same-proxy" type="b">
> > +            <default>true</default>
> > +            <summary>Unused; ignore</summary>
> > +            <description>
> > +            This key is not used, and should not be read or modified.
>
> can we remove it if it is not used?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/sugarlabs/sugar/pull/665/files/4bbac9afae7e02b854e02be492f163200a3f16b1#r62429966>
>